### PR TITLE
CMake: Cleanup include paths

### PR DESCRIPTION
**Pull request by chfast**
Fixes https://github.com/ethereum/solidity/issues/2956.

### CODE REVIEW

File: cmake/EthDependencies.cmake [hunk 1]
```diff
@@ -48,10 +48,3 @@ option(Boost_USE_STATIC_LIBS "Link Boost statically" ON)
 find_package(Boost 1.54.0 QUIET REQUIRED COMPONENTS regex filesystem unit_test_framework program_options system)
 
 eth_show_dependency(Boost boost)
-
-if (APPLE)
-	link_directories(/usr/local/lib)
-	include_directories(/usr/local/include)
```
Comment (axic) [inline cmake/EthDependencies.cmake:54]: This is the real fix for #2956, right?

Comment (chfast) [reply cmake/EthDependencies.cmake:54]: Not only. I also changed the order of deepening libs to make sure jsoncpp is included before boost (boost is installed in /usr/local/include).

**Comment by chfast** [comment]
The change about `<build dir>/include` maps changes made recently in cpp-ethereum: we generate headers in `<build dir>/include`. Before some were generated to to `<build dir>`, some to `<build dir>/include`.

**Review by axic — APPROVED** [review]
I don't understand cmake enough to judge this, but it doesn't seem to break anything and extra include paths are removed on the macOS so it probably fixes it.

**Comment by chfast** [comment]
Can you also review https://github.com/ethereum/cpp-ethereum/pull/4551 ?

**Pull request merged**